cpp/src/arrow/util/meson.build

```diff
@@ -0,0 +1,103 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+conf_data = configuration_data()
+
+conf_data.set('ARROW_VERSION_MAJOR', ver_major)
+conf_data.set('ARROW_VERSION_MINOR', ver_minor)
+conf_data.set('ARROW_VERSION_PATCH', ver_patch)
+
+conf_data.set('ARROW_VERSION', arrow_version)
+conf_data.set('ARROW_VERSION_STRING', arrow_version)
+
+conf_data.set('ARROW_SO_VERSION', arrow_so_version)
+conf_data.set('ARROW_FULL_SO_VERSION', arrow_full_so_version)
+
+# TODO: we may want to give Meson its own configuration file rather
```

Comment:
Instead of reusing the `config.h.cmake` file we would create one with directives specifically for Meson. That might make things slightly easier on the Meson side, although I'm not sure it is worth it to have two different config files

Reply:
We want to use `config.h.cmake` for both CMake and Meson (with `format: 'cmake@'`). It's for reducing maintenance cost.